### DISCUSSION TITLE
Copy to ROI not updating the original Mat

### DISCUSSION THREAD

**Discussion by MrZander** [emgucv/emgucv#451] [Q&A] [answered]
I am trying to resize (letterbox) an image before displaying it using the following:

```
UMat output = new UMat(size, DepthType.Cv8U, 4);
UMat resize = new UMat(1080, 1920, DepthType.Cv8U, 4);
UMat resizeSub = new UMat(360, 640, DepthType.Cv8U, 4);

using (var roi = new UMat(resize, new System.Drawing.Rectangle(scaledRoiX, scaledRoiY, scaledWidth, scaledHeight))) 
{
	CvInvoke.Resize(output, roi, new System.Drawing.Size(0, 0), scaleFactor, scaleFactor);
	CvInvoke.Resize(roi, resizeSub, resizeSub.Size);
        CvInvoke.ImShow("roi", roi);
}

CvInvoke.ImShow("output", output);
CvInvoke.ImShow("resize", resize);
CvInvoke.ImShow("resizeSub", resizeSub);
```

The result is:
`roi` is fine and displays the scaled image
`output` is fine, its the full sized image
`resizeSub` is fine, its a scaled version of the output
`resize` is the correct size, but is pure black.

The question is, how can `roi` contain my image data, but not `resize`?  `roi` is supposed to directly modify the parent mat, correct?

What am I doing wrong?

**Comment by emgucv** [comment, marked as answer]
When you call:
`CvInvoke.Resize(output, roi, new System.Drawing.Size(0, 0), scaleFactor, scaleFactor);`
The newly computed size of roi is (output.width/scaleFactor, output.height/scaleFactor). If it does not match the existing roi.Size (scaledWidth, scaledHeight), "roi" will be re-allocated with new memory meeting the new size requirement, and therefore no longer share data with the parent.

**Reply by MrZander** [reply]
`scaleFactor` is calculated based on the output and desired ROI size, so that isn't it.
But, this did lead me down the right path.  The issue was further up the line, output was getting converted from Bgra to Bgr, so the difference in the number of channels was causing the reallocation.

Thanks!
